### PR TITLE
Add series of unit tests

### PR DESCRIPTION
**Pull request by azakordonets**
## Proposed changes
Mostly unit tests. For some commands classes introduced custom constructors to be able to mock dependencies. `Commands` now covered for about 80% . Also covered some parts of `collections`, `impl`, `conditions`,  `logevents` packages. 
## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome htmlunit` command
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/test/java/com/codeborne/selenide/commands/SelectOptionContainingTextTest.java [hunk 1]
```diff
@@ -44,11 +55,22 @@ public void selects_allMatchingOptions_forMultipleSelect() {
...
+          By.xpath(".//option[contains(normalize-space(.), \"option-subtext\")]"));
+
+    command.execute(proxy, select, new Object[] {"option-subtext"});
 
     verify(option1).click();
     verify(option2).click();
   }
+
+  @Test
+  public void throws_NoSuchElementException_whenNoElementsFound() {
```
Comment (rosolko) [inline src/test/java/com/codeborne/selenide/commands/SelectOptionContainingTextTest.java]: Small note - can you please remove `_` chars for test name?

File: src/test/java/com/codeborne/selenide/commands/IsDisplayedCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  public void testExecuteMethodWhenExceptionWithInvalidSelectorException() {
+    catchExecuteMethodWithException(new NotFoundException("invalid selector"));
+  }
+
+  private <T extends Throwable> void catchExecuteMethodWithException(T exception) {
+    when(locator.getWebElement()).thenReturn(mockedElement);
+    doThrow(exception).when(mockedElement).isDisplayed();
+    assertFalse(isDisplayedCommand.execute(proxy, locator, new Object[] {"something more"}));
+  }
+
```
Comment (rosolko) [inline src/test/java/com/codeborne/selenide/commands/IsDisplayedCommandTest.java]: Please remove this empty line

File: src/test/java/com/codeborne/selenide/commands/IsDisplayedCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
```
Comment (rosolko) [inline src/test/java/com/codeborne/selenide/commands/IsDisplayedCommandTest.java]: And this

File: src/test/java/com/codeborne/selenide/commands/SelectOptionByTextOrIndexCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    selectOptionByTextOrIndexCommand = new SelectOptionByTextOrIndex();
+    proxy = mock(SelenideElement.class);
+    selectField = mock(WebElementSource.class);
+    mockedElement = mock(WebElement.class);
+    mockedFoundElement = mock(WebElement.class);
+
+    when(selectField.getWebElement()).thenReturn(mockedElement);
+    when(mockedElement.getText()).thenReturn(defaultElementText);
+    when(mockedElement.getTagName()).thenReturn("select");
+
```
Comment (rosolko) [inline src/test/java/com/codeborne/selenide/commands/SelectOptionByTextOrIndexCommandTest.java]: I think there are no sens to split when lines with empty line

File: src/test/java/com/codeborne/selenide/commands/SetSelectedCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SetSelectedCommandTest {
+  private SelenideElement proxy;
+  private WebElementSource locator;
+  private SetSelected setSelectedCommand;
+  private WebElement mockedFoundElement;
+
```
Comment (rosolko) [inline src/test/java/com/codeborne/selenide/commands/SetSelectedCommandTest.java]: Remove one line please

File: src/test/java/com/codeborne/selenide/commands/ShouldBeCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ShouldBeCommandTest {
+  private SelenideElement proxy;
+  private WebElementSource locator;
+  private ShouldBe shouldBeCommand;
+  private WebElement mockedFoundElement;
+
```
Comment (rosolko) [inline src/test/java/com/codeborne/selenide/commands/ShouldBeCommandTest.java]: And here

File: src/test/java/com/codeborne/selenide/commands/ShouldHaveCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ShouldHaveCommandTest {
+  private SelenideElement proxy;
+  private WebElementSource locator;
+  private ShouldHave shouldHaveCommand;
+  private WebElement mockedFoundElement;
+
```
Comment (rosolko) [inline src/test/java/com/codeborne/selenide/commands/ShouldHaveCommandTest.java]: And here

File: src/test/java/com/codeborne/selenide/commands/ToStringCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    assertEquals("Element not found {By.name: }", elementString);
+  }
+
+  @Test
+  public void testExecuteMethodWhenIndexOutOfBoundExceptionIsThrown() {
+    doThrow(new IndexOutOfBoundsException()).when(locator).getWebElement();
+    String elementString = toStringCommand.execute(proxy, locator, new Object[] {});
+    assertEquals("java.lang.IndexOutOfBoundsException", elementString);
+  }
+
```
Comment (rosolko) [inline src/test/java/com/codeborne/selenide/commands/ToStringCommandTest.java]: And here

**Comment by BorisOsipov** [comment]
@azakordonets Thank you for another awesome PR. Your help with tests is beyond price 👍

**Comment by rosolko** [comment]
@azakordonets Many thanks for PR 🏆

**Comment by azakordonets** [comment]
Awesome, glad that i could help :)

**Pull request merged**